pytorch_lightning/plugins/checkpoint/torch.py

```diff
@@ -0,0 +1,27 @@
+from pathlib import Path
+from typing import Any, Dict, Optional, Union
+
+import pytorch_lightning as pl
+from pytorch_lightning.plugins.checkpoint.checkpoint import CheckpointIOPlugin, TLoadStorageOptions, TSaveStorageOptions
+from pytorch_lightning.utilities import rank_zero_warn
+from pytorch_lightning.utilities.cloud_io import atomic_save
+from pytorch_lightning.utilities.cloud_io import load as pl_load
+
+
+class TorchCheckpointIOPlugin(CheckpointIOPlugin):
+    def save_checkpoint(
+        self, checkpoint: Dict[str, Any], path: Union[str, Path], storage_options: Optional[TSaveStorageOptions] = None
+    ) -> None:
+        try:
+            # write the checkpoint dictionary on the file
+            atomic_save(checkpoint, path)
+        except AttributeError as err:
+            key = pl.LightningModule.CHECKPOINT_HYPER_PARAMS_KEY
```

Comment:
As discussed @ananthsub I think the logic here is to remove the hparams if a key exists that causes it to crash, with the original PR from way back suggesting it was added to handle some pickling issue from DDP. In a separate PR we should remove this and see what tests fail!